Automattic/pocket-casts-android
# Migrate FileUtil from Java to Kotlin

## Description

Continuation of Java migration outlined in #44. This migrates [FileUtil.java](https://github.com/Automattic/pocket-casts-android/blob/61fe9bf02411dbb158ec838d2fc4834099e99a4b/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/FileUtil.java) and deletes [StringUtil.java](https://github.com/Automattic/pocket-casts-android/blob/61fe9bf02411dbb158ec838d2fc4834099e99a4b/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/StringUtil.java) as `FileUtil` was the last place that used it.

The main changes are making `copyFile()` and `copyDir()` functions private. The only entry point is now `copy()` function that delegated to the mentioned ones anyway.

## Testing Instructions

Code is covered by tests so the migration doesn't have to be tested extensively. In general, you should check if file management works as expected.

1. Download a podcast episode.
2. Share it and see if `Open file in…` still works.
3. Delete the downloaded episode.
4. Downloaded couple of other episodes.
5. In the `Storage & data use` settings clear downloaded files.
6. Verify that downloaded files no longer exist.

## Checklist
 
#### I have tested any UI changes...
